snowplow-tracker/src/main/java/com/snowplowanalytics/snowplow/tracker/Tracker.java

```diff
@@ -485,6 +485,11 @@ public void track(final Event event) {
             return;
         }
 
+        if (ScreenView.class.isInstance(event) && screenState != null) {
```

Comment:
Not for this PR but just a comment that when the refactor is underway, I'd love to see all this type checking removed. Makes me a little sad everytime I stumble across it.

Reply:
Yeah same feeling for me!